hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestConnectionManager.java

```diff
@@ -301,4 +301,16 @@ public void testConfigureConnectionActiveRatio() throws IOException {
 
     tmpConnManager.close();
   }
+
+  @Test
+  public void testUnsupportedProtoExceptionMsg() throws IOException {
+    exceptionRule.expect(IllegalStateException.class);
+    exceptionRule
+        .expectMessage("Unsupported protocol for connection to NameNode: "
+            + UnsupportedProto.class.getName());
+    ConnectionPool.newConnection(conf, TEST_NN_ADDRESS, TEST_USER1,
+        UnsupportedProto.class);
+  }
+
+  interface UnsupportedProto { }
```

Comment:
Instead Can you LambdaTestUtils, Like this :
` @Test
  public void testUnsupportedProtoExceptionMsg() throws Exception {
    LambdaTestUtils.intercept(IllegalStateException.class,
        "Unsupported protocol for connection to NameNode: "
            + TestConnectionManager.class.getName(),
        () -> ConnectionPool.newConnection(conf, TEST_NN_ADDRESS, TEST_USER1,
            TestConnectionManager.class));
  }`

Reply:
Got it.